var/spack/repos/builtin/packages/kentutils/package.py

```diff
@@ -7,35 +7,138 @@
+    variant(
+        "htslib",
+        default=True,
+        description="Build with bundled htslib (using an external htslib may lead to errors)",
+        sticky=True,
+    )
+
+    with default_args(type=("build", "link", "run")):
+        depends_on("libpng")
+        depends_on("openssl")
+        depends_on("uuid")
+        depends_on("mysql-client")
+        depends_on("zlib-api")
+        depends_on("freetype")
+        depends_on("libiconv")
+        depends_on("htslib+pic", when="~htslib+libs")
+
+    # The bgzip.c bug present in other packages is present in kent/src/htslib/bgzf.c
+    # Conflicting line: assert(compressBound(BGZF_BLOCK_SIZE) < BGZF_MAX_BLOCK_SIZE);
+    # We can patch this by removing the assertion, but there are still performance issues
+    # See: https://github.com/samtools/htslib/issues/1257
+    conflicts("zlib-ng")
 
-    requires("%gcc", when="@302.1")
+    # Does not add a link to mysql_config, which is required for compilation
+    conflicts("mariadb-c-client")
+
+    # MySQL pointer/integer conversion issue
+    patch("fix-mysql-options-gcc13.patch", when="%gcc@13: ^mysql")
+    # MySQL build flags improperly states the zlib-api library
+    patch("mysql-zlib-workaround.patch", when="%gcc ^mysql")
 
     def flag_handler(self, name, flags):
         if name == "ldflags":
             flags.append(f'{self.spec["libiconv"].libs.ld_flags}')
+        elif name == "cflags" and self.spec.satisfies("+libs"):
+            flags.append("-fPIC")
         return (flags, None, None)
 
+    @property
+    def machtype(self):
+        # This is hard-coded in the Makefile and included here for reference
+        # and to make it adjustable if we need to adjust this in the future
+        return "local"
+
+    @property
+    def headers(self):
+        headers = []
+        if self.spec.satisfies("+libs"):
+            headers.extend(find_headers("*", self.prefix.inc, recursive=True))
+        if self.spec.satisfies("+htslib+libs"):
+            headers.extend(find_headers("*", self.prefix.htslib, recursive=True))
+        return HeaderList(headers)
+
+    @property
+    def libs(self):
+        return LibraryList([join_path(self.prefix, lib) for lib in self.local_libs])
+
+    @property
+    def local_libs(self):
+        libs = []
+        if self.spec.satisfies("+libs"):
+            libs.extend(
+                [
+                    f"{self.machlib}/jkweb.a",
+                    f"{self.machlib}/jkOwnLib.a",
+                    f"{self.machlib}/jkhgap.a",
+                    f"{self.machlib}/jkhgapcgi.a",
+                    f"hg/altSplice/{self.machlib}/libSpliceGraph.a",
+                ]
+            )
+        if self.spec.satisfies("+htslib+libs"):
+            libs.append("htslib/libhts.a")
+        return LibraryList(libs)
+
+    @property
+    def machlib(self):
+        return f"lib/{self.machtype}"
+
+    @property
+    def lib_dir(self):
+        return join_path(self.prefix, self.machlib)
```

Comment:
Is this property used anywhere?

Reply:
It's used in dependent package, but might be better handled in setup_dependent_package.